src/Plivo/Resource/Lookup/LookupInterface.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Plivo.Client;
+
+
+namespace Plivo.Resource.Lookup
+{
+    public class LookupInterface : ResourceInterface
+    {
+        public LookupInterface(HttpClient client) : base(client)
+        {
+            Uri = "Lookup/Number/";
+        }
+
+        // getResource is like the method 'GetResource' from parent class but
+        // doesn't add a trailing slash at the end.
+        private async Task<T> getResource<T>(string resource, Dictionary<string, object> data = null)
+        where T : new()
+        {
+            var result = await Client.Fetch<T>(Uri + resource, data);
+            try
+            {
+                result.Object.GetType().GetRuntimeProperty("StatusCode").SetValue(result.Object, result.StatusCode, null);
+            }
+            catch (System.NullReferenceException)
+            {
```

Comment:
Only NullReferenceException ? any plan to handle Generic exception?

Reply:
Other exceptions will be propagated to the caller. Didn't want to change existing behaviour. This is the code from parent class: https://github.com/plivo/plivo-dotnet/blob/master/src/Plivo/Resource/ResourceInterface.cs#L52-L54